### PR TITLE
Add batched insert for the initial import and full PostgreSQL support

### PR DESCRIPTION
**Pull request by alexsporn**
- This change increases the speed of the initial import by parallelizing the processing of the unspent outputs.
- This also adds full support for using PostgreSQL as an alternative to SQLite for enhanced performance with large datasets.

### CODE REVIEW

File: core/indexer/component.go [hunk 1]
```diff
@@ -240,45 +270,100 @@ func checkIndexerStatus(ctx context.Context) (*indexer.Status, error) {
...
-		CoreComponent.LogInfof("> Indexer started at ledgerIndex %d", indexerStatus.LedgerIndex)
+		CoreComponent.LogInfo("Checking database schema")
+		if err := deps.Indexer.AutoMigrate(); err != nil {
+			return nil, err
+		}
+		CoreComponent.LogInfof("> Indexer started at ledgerIndex %d", status.LedgerIndex)
 	}
 
-	return indexerStatus, nil
+	// Run auto migrate to make sure all required tables and indexes are there
```
Comment (muXxer) [inline core/indexer/component.go]: is the comment in the correct line here?

File: pkg/database/engine.go [hunk 2]
```diff
@@ -197,27 +212,42 @@ func (l *sqliteLogger) Printf(t string, args ...interface{}) {
...
-func NewWithDefaultSettings(path string, createDatabaseIfNotExists bool, log *logger.Logger) (*gorm.DB, error) {
+func NewWithDefaultSettings(dbParams Params, createDatabaseIfNotExists bool, log *logger.Logger) (*gorm.DB, Engine, error) {
 
-	targetEngine, err := CheckEngine(path, createDatabaseIfNotExists, EngineSQLite)
-	if err != nil {
-		return nil, err
+	targetEngine := dbParams.Engine
+	if len(dbParams.Path) > 0 {
+		var err error
+		targetEngine, err = CheckEngine(dbParams.Path, createDatabaseIfNotExists, dbParams.Engine)
```
Comment (muXxer) [inline pkg/database/engine.go]: Add SQLite

File: pkg/indexer/indexer.go [hunk 3]
```diff
@@ -345,13 +365,11 @@ func (i *Indexer) Status() (*Status, error) {
 
 func (i *Indexer) Clear() error {
 	// Drop all tables
-	for _, table := range tables {
-		if err := i.db.Migrator().DropTable(table); err != nil {
-			return err
-		}
+	if err := i.db.Migrator().DropTable(dbTables...); err != nil {
```
Comment (muXxer) [inline pkg/indexer/indexer.go:368]: maybe rename to `DropTables`?

Comment (alexsporn) [reply pkg/indexer/indexer.go:368]: This is not my api, it's from gorm

Comment (muXxer) [reply pkg/indexer/indexer.go:368]: shame on them!

**Review by muXxer — APPROVED** [review]

**Pull request merged**